external_search.py

```diff
@@ -80,13 +80,30 @@ def set_works_index_and_alias(self, current_alias):
         """Finds or creates the works_index and works_alias based on
         provided configuration.
         """
-        index_details = self.indices.get_alias(name=current_alias, ignore=[404])
-        found = not (index_details.get('status')==404 or 'error' in index_details)
+        try:
+            index_details = self.indices.get(current_alias)
+            found_index = True
+        except Exception, e:
+            found_index = False
+
+        if found_index:
+            # The alias exists as an index. This isn't supposed to happen,
+            # but we can avoid a meltdown now by doing nothing.
+            self.log.error(
+                'Supposed alias "%s" already exists, as an index.',
+                current_alias
+            )
+            return
```

Comment:
That's a problem. It works when I run it for real (not sure how) but it makes tests fail.